tests/test_on_data/test_gin_ecephys.py

```diff
@@ -235,34 +235,50 @@ class TestConverter(NWBConverter):
     )
     def test_neuroscope_gains(self, name, conversion_options):
         input_gain = 2.0
-        data_interface = NeuroscopeRecordingInterface
         interface_kwargs = dict(file_path=str(DATA_PATH / "neuroscope" / "test1" / "test1.dat"), gain=input_gain)
 
-        nwbfile_path = str(self.savedir / f"{data_interface.__name__}-{name}.nwb")
+        nwbfile_path = str(self.savedir / f"test_neuroscope_gains_{name}.nwb")
 
         class TestConverter(NWBConverter):
-            data_interface_classes = dict(TestRecording=data_interface)
+            data_interface_classes = dict(TestRecording=NeuroscopeRecordingInterface)
 
         converter = TestConverter(source_data=dict(TestRecording=interface_kwargs))
         converter.run_conversion(nwbfile_path=nwbfile_path, overwrite=True, conversion_options=conversion_options)
 
-        # nwb file check-test
-        io = NWBHDF5IO(nwbfile_path, "r")
-        nwbfile_in = io.read()
-        output_conversion = nwbfile_in.acquisition["ElectricalSeries_raw"].conversion
-        output_gain = output_conversion * 1e6
-        assert input_gain == pytest.approx(output_gain)
+        with NWBHDF5IO(path=nwbfile_path, mode="r") as io:
+            nwbfile_in = io.read()
+            output_conversion = nwbfile_in.acquisition["ElectricalSeries_raw"].conversion
+            output_gain = output_conversion * 1e6
+            self.assertEqual(first=input_gain, second=output_gain)
```

Comment:
```suggestion
            self.assertAlmostEqual(first=input_gain, second=output_gain)
```

Reply:
Is exact equal failing for some reason? I can't think of a reason they would actually change at the floating point level, and the tests seem to be passing as is

In general, stronger statements are better when you can get away with them. For certain array testing (like the trace calls), we use approximate testing because array equality is calculated on a matrix norm, which can easily incur very minor epsilon's from floating point errors accolated across all the elements.

Reply:
I have this idea that as a general practice float equality should be tested almost equal (epsilon difference) because of floating point issues.

Reply:
For data quality, like trace scaling through SpikeInterface readers, I'd completely agree. But for metadata fields (attributes, column values, etc) I do think exact representation matching ought to be the ideal whenever it's achievable.

Explicit example: in Neuroscope, when writing gains, if the input `gain_to_uV=0.195`, the NWBFile ought to store `nwbfile.acquisition["ElectricalSeries_raw"].conversion` as `0.195E-6` instead of `0.1944444449E-6`, unless there's no way around it. But here it seems to be passing exactness anyhow.

Reply:
All right, that makes sense.